tiledb/sm/filesystem/s3.cc

```diff
@@ -223,7 +238,7 @@ Status S3::disconnect() {
           make_multipart_complete_request(state);
       auto outcome = client_->CompleteMultipartUpload(complete_request);
       if (!outcome.IsSuccess()) {
-        return LOG_STATUS(Status::S3Error(
+        st = LOG_STATUS(Status::S3Error(
```

Comment:
I wouldn't worry about it now, but here an error `st` may be overwritten by a success `st` later, which means this function would return success.

Reply:
Done, good catch